scapy/contrib/automotive/uds_scan.py

```diff
@@ -297,9 +299,54 @@ def post_execute(self, socket, state, global_configuration):
 class UDS_RDBIEnumerator(UDS_Enumerator):
     _description = "Readable data identifier per state"
 
+    @staticmethod
+    def _random_results_to_scan_range(rdbi_random, state):
+        # type: (UDS_RDBIRandomEnumerator, EcuState) -> Iterable[int]
+
+        identifiers_with_positive_response = [p.resp.dataIdentifier
+                                              for p in rdbi_random.results_with_positive_response
+                                              if p.state == state]
+
+        if len(identifiers_with_positive_response) == 0:
+            # quick path for better performance
+            scan_range = []
+        else:
+            def flatten(l):
+                """[[s1, s2, s3], [s4, s5]] -> [s1, s2, s3, s4, s5]"""
+                return [item for sublist in l for item in sublist]
+
+            # by random enumerator
+            probed_identifiers = \
+                set(flatten([p.req.identifiers
+                    for p in rdbi_random.results
+                    if p.state == state]))
+
+            block_size = UDS_RDBIRandomEnumerator.block_size
+            to_scan = []
+            for block_index, start in enumerate(range(0, 2 ** 16, block_size)):
+                end = start + block_size - 1
+                pr_in_block = any((start <= identifier <= end
+                                   for identifier in identifiers_with_positive_response))
+                if pr_in_block:
+                    to_scan += range(start, end + 1)
```

Comment:
This function might be useful: https://docs.python.org/3/library/itertools.html#itertools.chain

Reply:
At the time of calling `itertools.chain` the lists have already to be known. In my code, it's not known at this point. That's why it's not practicle here